bin/utils.dart

```diff
@@ -0,0 +1,33 @@
+import 'dart:async';
+import 'dart:io';
+import 'package:io/io.dart';
+
+class PresubmitUtils {
+  ProcessManager manager;
+
+  PresubmitUtils() {
+    this.manager = new ProcessManager();
+  }
+
+  /// Fetches the configlet file if it doesn't exist already.
+  Future<Null> fetchConfiglet() async {
+    File configletFile = new File('bin/configlet');
+
+    if (!configletFile.existsSync()) {
+      print('Fetching configlet...');
+      Process spawn = await manager.spawn('bin/fetch-configlet', []);
+      await spawn.exitCode;
+    }
+  }
+
+  /// Runs the [executable] with [arguments].
+  Future<Null> run(String executable, [List<String> arguments = const []]) async {
```

Comment:
Renaming `run` to `runCmd`  and changing the return type to `Future<int>` can also prove useful. Such as in our create-exercise script #67 we are using `Process.run`. The name `runCmd` clearly states its going to run a command, Once you rename that to `runCmd` we can reuse your code in other scripts too. Other scripts can call this function and if the `Future` resolves can do `something` and if it fails can do `something on fail`. They don't have to worry about the `exitCode` or outputting the process output, they get it for free. Similar to `fetchConfiglet` return type needs to be `Future<int>` so the `exitCode` can be utilized by the calling script. 
FYI spawn.exitCode is a `Future<int>`

Reply:
You will need to use `FutureOr<int>` instead of `Future<int>` if you chose to add a check for `isExecutable`.